libs/server/Storage/Session/MainStore/MainStoreOps.cs

```diff
@@ -1190,5 +1193,299 @@ public GarnetStatus MemoryUsageForKey<TContext, TObjectContext>(ArgSlice key, ou
 
             return status;
         }
+
+        public unsafe GarnetStatus LCS(ArgSlice key1, ArgSlice key2, ref SpanByteAndMemory output, bool lenOnly = false, bool withIndices = false, bool withMatchLen = false, int minMatchLen = 0)
+        {
+            var createTransaction = false;
+            if (txnManager.state != TxnState.Running)
+            {
+                txnManager.SaveKeyEntryToLock(key1, false, LockType.Shared);
+                txnManager.SaveKeyEntryToLock(key2, false, LockType.Shared);
+                txnManager.Run(true);
+                createTransaction = true;
+            }
+
+            var context = txnManager.LockableContext;
+            try
+            {
+                var status = LCSInternal(key1, key2, ref output, ref context, lenOnly, withIndices, withMatchLen, minMatchLen);
+                return status;
+            }
+            finally
+            {
+                if (createTransaction)
+                    txnManager.Commit(true);
+            }
+        }
+
+        private unsafe GarnetStatus LCSInternal<TContext>(ArgSlice key1, ArgSlice key2, ref SpanByteAndMemory output, ref TContext context, bool lenOnly = false, bool withIndices = false, bool withMatchLen = false, int minMatchLen = 0)
+            where TContext : ITsavoriteContext<SpanByte, SpanByte, RawStringInput, SpanByteAndMemory, long, MainSessionFunctions, MainStoreFunctions, MainStoreAllocator>
+        {
+            var isMemory = false;
+            MemoryHandle ptrHandle = default;
+            var ptr = output.SpanByte.ToPointer();
+            var curr = ptr;
+            var end = curr + output.Length;
+
+            try
+            {
+                if (lenOnly && withIndices)
```

Comment:
Can do this check in ArrayCommands, so that you don't incur the cost of starting a transaction

Reply:
I have moved it to ArrayCommands but I thought the methods in IGarnetApi could be called programmatically, so I thought we should have command-specific validation inside the Ops

Reply:
That's correct, we usually assume input correctness when called programmatically.